Directory.Build.targets

```diff
@@ -1,5 +1,138 @@
 <Project>
-  <PropertyGroup Condition="'$(PlatformTarget)' == 'x86' And '$(TargetFrameworkIdentifier)' != '.NETFramework'">
-    <RunCommand>$(MSBuildProgramFiles32)\dotnet\dotnet</RunCommand> <!-- Support WindowsXNA 32bit debugging in VS -->
-  </PropertyGroup>
+  <Target Name="ShowBuildInfo" BeforeTargets="CoreCompile" Condition="'$(MSBuildProjectName)' == 'DXMainClient'">
+    <Message Importance="high" Text="Engine: $(Engine); Game: $(Game); Platform: $(Platform); TargetFramework: $(TargetFramework); Configuration: $(Configuration)" />
+  </Target>
+
+  <Target Name="MakeDirectoryStructure" AfterTargets="Publish" Condition="'$(MSBuildProjectName)' == 'DXMainClient'">
+    <CallTarget Targets="PublishNetFrameworkWindowsGLNative;RemoveNetFrameworkWindowsGLNative;RemoveWindowsDXNonWindowsBinaries;RemoveWindowsGLNonWindowsBinaries" />
+    <CallTarget Targets="MoveCommonBinaries;MoveCommonBinariesNetFx;MoveClientExes" Condition="'$(NoMove)' != 'true'" />
+  </Target>
+
+  <Target Name="PublishNetFrameworkWindowsGLNative" Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework' And '$(Engine)' == 'WindowsGL'">
+    <Message Importance="high" Text="Copying NetFramework WindowsGLNative files" />
+    <ItemGroup>
+      <_lib_x64 Include="$(OutputPath)\x64\*.*" />
+      <_lib_x86 Include="$(OutputPath)\x86\*.*" />
+    </ItemGroup>
+    <Copy SourceFiles="@(_lib_x64)" DestinationFolder="$(PublishDir)\x64" />
+    <Copy SourceFiles="@(_lib_x86)" DestinationFolder="$(PublishDir)\x86" />
+  </Target>
+
+  <Target Name="RemoveNetFrameworkWindowsGLNative" AfterTargets="PublishNetFrameworkWindowsGLNative">
+    <Message Importance="high" Text="Removing not needed NetFramework WindowsGLNative files" />
+    <Delete Files="$(PublishDir)SDL2.dll" />
+    <Delete Files="$(PublishDir)soft_oal.dll" />
+  </Target>
+
+  <Target Name="RemoveWindowsDXNonWindowsBinaries" Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows' And '$(Engine)' == 'WindowsDX'">
+    <Message Importance="high" Text="Removing unnecessary WindowsDX files" />
+    <RemoveDir Directories="$(PublishDir)runtimes/debian-x64" />
+    <RemoveDir Directories="$(PublishDir)runtimes/fedora-x64" />
+    <RemoveDir Directories="$(PublishDir)runtimes/opensuse-x64" />
+    <RemoveDir Directories="$(PublishDir)runtimes/osx" />
+    <RemoveDir Directories="$(PublishDir)runtimes/rhel-x64" />
+  </Target>
+
+  <Target Name="RemoveWindowsGLNonWindowsBinaries" Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows' And '$(Engine)' == 'WindowsGL'">
+    <Message Importance="high" Text="Removing unnecessary WindowsGL files" />
+    <RemoveDir Directories="$(PublishDir)runtimes/linux-x64" />
+    <RemoveDir Directories="$(PublishDir)runtimes/osx" />
+  </Target>
+
+  <!-- Note: the folder structure should be consistent with ./build.ps1 file -->
+  <Target Name="MoveCommonBinaries">
```

Comment:
> Since you have updated packages, could you update this common dll lists? or perhaps there are no differences?

Do we have some tools to easily calculate the list of files?

Reply:
currently no